pkg/syncer/util.go

```diff
@@ -105,7 +105,12 @@ func IsValidVolume(ctx context.Context, volume v1.Volume, pod *v1.Pod, metadataS
 	}
 
 	// Verify if pv is vsphere csi volume
-	if (pv.Spec.CSI == nil || pv.Spec.CSI.Driver != csitypes.Name) && (metadataSyncer.coCommonInterface.IsFSSEnabled(ctx, common.CSIMigration) && pv.Spec.VsphereVolume == nil) {
+	if pv.Spec.CSI == nil || pv.Spec.CSI.Driver != csitypes.Name {
+		// pv.Spec.CSI can be nil for VCP volumes. Verify if migration is enabled & check if the volume is a valid migrated VCP volume
+		if metadataSyncer.coCommonInterface.IsFSSEnabled(ctx, common.CSIMigration) && pv.Spec.VsphereVolume != nil && !isValidvSphereVolume(ctx, pv.ObjectMeta) {
+			log.Debugf("Pod %s has a valid vSphereVolume but the volume is not migrated", pod.Name)
```

Comment:
from this log message, we can not determine if FSS was disabled or volume was not a vsphere volume

Reply:
Line: 118 below  already has that log message

Reply:
As discussed I have refactored the checks. PTAL